### PR TITLE
 Refactoring to filter next appointment date

### PR DESCRIPTION
**Pull request by Marymary-dev**

### CODE REVIEW

**Comment by nobert-mumo** [comment]
@Marymary-dev I was wondering, are we going to also implement the data management processes Ashley was also using? 
e.g restricting to last 6 months, imputing missing dates and dups etc.

**Comment by cbrianbet** [comment]
Hi @nobert-mumo @Marymary-dev I see this PR is 2 months old 
Is it still valid?

**Comment by nobert-mumo** [comment]
> Hi @nobert-mumo @Marymary-dev I see this PR is 2 months old Is it still valid?

Hi @Marymary-dev also co-asking :) Did we find out if the filtering to the last 6 months aAsOfdate is still necessary ?

File: Scripts/ODS/utils/load_HistoricalAppointmentStatus.sql [hunk 1]
```diff
@@ -214,16 +221,16 @@ last_exit_as_of_date as (
 Select  [DateRecieved],ROW_NUMBER()OVER(Partition by Sitecode Order by [DateRecieved] Desc) as Num ,
 	SiteCode,
     cast( [DateRecieved]as date) As DateReceived
-from DWAPICentral.dbo.FacilityManifest 
+from ODS.dbo.CT_FacilityManifest
```
Comment (nobert-mumo) [inline Scripts/ODS/utils/load_HistoricalAppointmentStatus.sql:224]: nice!

File: Scripts/ODS/utils/load_HistoricalAppointmentStatus.sql [hunk 2]
```diff
@@ -368,18 +375,19 @@ ARTOutcomesCompuation as (
...
     last_upload_as_of_date.DateRecieved 
 from ARTOutcomesCompuation  
-left join last_exit_as_of_date on last_exit_as_of_date.PatientIDHash=ARTOutcomesCompuation.PatientIDHash
-and last_exit_as_of_date.PatientPK= ARTOutcomesCompuation.PatientPK
+left join last_exit_as_of_date on  last_exit_as_of_date.PatientPK= ARTOutcomesCompuation.PatientPK
 and last_exit_as_of_date.sitecode=ARTOutcomesCompuation.sitecode
  left  join  last_upload_as_of_date on  last_upload_as_of_date.SiteCode=ARTOutcomesCompuation.SiteCode 
-    where  NextAppointmentDate > LastencounterDate 
+               WHERE ARTOutcomesCompuation.NextAppointmentDate > ARTOutcomesCompuation.LastEncounterDate
+   -- AND ARTOutcomesCompuation.NextAppointmentDate > DATEADD(month, -6, ARTOutcomesCompuation.AsOfDate)
```
Comment (nobert-mumo) [inline Scripts/ODS/utils/load_HistoricalAppointmentStatus.sql:382]: @Marymary-dev  maybe we can get rid of the commented out code ?

File: Scripts/ODS/utils/load_HistoricalAppointmentStatus.sql [hunk 3]
```diff
@@ -160,7 +165,9 @@ exits_as_of_date as (
         ExitReason,
 		ReEnrollmentDate
     from ODS.dbo.CT_PatientStatus
-    where ExitDate <= @as_of_date 
+    where  ExitDate <= @as_of_date
+    --ExitDate>= DATEADD(month, -6, @as_of_date)
```
Comment (nobert-mumo) [inline Scripts/ODS/utils/load_HistoricalAppointmentStatus.sql:169]: here also..

File: Scripts/ODS/utils/load_HistoricalAppointmentStatus.sql [hunk 4]
```diff
@@ -76,7 +81,7 @@ with clinical_visits_as_of_date as (
     from ODS.dbo.CT_ARTPatients
     left join ODS.dbo.CT_Patient  on  CT_Patient.PatientPK = CT_ARTPatients.PatientPK
     and CT_Patient.SiteCode = CT_ARTPatients.SiteCode
-	where  ODS.dbo.CT_ARTPatients.SiteCode > 0 and ODS.dbo.CT_ARTPatients.LastVisit <= @as_of_date 
+	where  ODS.dbo.CT_ARTPatients.SiteCode > 0 and ODS.dbo.CT_ARTPatients.ExpectedReturn <=@as_of_date
```
Comment (nobert-mumo) [inline Scripts/ODS/utils/load_HistoricalAppointmentStatus.sql:84]: @Marymary-dev  just to understand here for ARTPatients why the change to filter on ExpectedReturn ?
For visits and pharamacy above I get it.

Comment (nobert-mumo) [reply Scripts/ODS/utils/load_HistoricalAppointmentStatus.sql:84]: @Marymary-dev oh yeah, I meant here on `CT_ARTPatients`

**Comment by nobert-mumo** [comment]
@Marymary-dev LGTM. I think something happened on the diffs but I see the review comments were addressed.

**Review by nobert-mumo — APPROVED** [review]
